app/controllers/concerns/clear_wizard_cache.rb

```diff
@@ -0,0 +1,26 @@
+module ClearWizardCache
+  include ActiveSupport::Concern
+
+private
+
+  def clear_wizard_if_new_entry(wizard, entrypoint_check: false)
+    return if (entrypoint_check && !entrypoint_path?) ||
+              (!entrypoint_check && entrypoint_in_wizard_flow?(wizard_flow_controllers, wizard_controller_excluded_paths))
```

Comment:
What's the reason we have two options here?

I think the entrypoint_path one makes sense, where we specify which paths are the entrypoints, and clear the wizard when we come from them. I don't really understand the other one though

Reply:
Ok, discussed in person.

The way we wanted to do this is to specify a controller (or list of controllers) that make up the wizard flow, and reset the wizard if the referrer is not from the specified controllers.

The first caveat is that not all actions in wizard controllers are part of the flow e.g. `interviews#index` is not in the interview edit or create flow. Therefore we add  a list of paths which are from the specified controllers but not in the wizard flows.

The second caveat is that if routes are not set up in a particular way using `resources`, then this controller detection method does not work. So for more complicated controller setups, like reasons for rejection, we just specify which routes are the entrypoints for the flow, and clear the wizard if the referrer is one of those.